provider/manual/provider.go

```diff
@@ -57,6 +60,35 @@ func (p manualProvider) CloudSchema() *jsonschema.Schema {
 	return cloudSchema
 }
 
+// Ping tests the connection to the cloud, to verify the endpoint is valid.
+func (p manualProvider) Ping(endpoint string) error {
+	// do a normal TCP dial first, because it makes the error messages clearer
+	// when things fail.
+	addr := net.JoinHostPort(endpoint, "22")
+	c, err := net.Dial("tcp", addr)
+	if err != nil {
+		return errors.Errorf("Can't connect to %s: "+err.Error(), addr)
+	}
+	c.Close()
+
+	// Now that we know we can connect, do the actual connection attempt.
+	// There's no "just connect" command for utils/ssh, so we run a command that
+	// should always work.
+	cmd := ssh.Command(endpoint, []string{"echo", "hi"}, nil)
```

Comment:
I would talk with some more linuxy people about this. There is probably a safer/more minimal command to issue.

Also, does the `ssh` package not have some kind of analogue to `Ping`? Do we need to run a command?